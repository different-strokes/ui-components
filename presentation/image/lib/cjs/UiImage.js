'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var UiImage = function UiImage(_ref) {
  var alt = _ref.alt,
      caption = _ref.caption,
      src = _ref.src,
      theme = _ref.theme;
  return _react2.default.createElement(
    'figure',
    { className: theme.imageContainer },
    _react2.default.createElement('img', { src: src, alt: alt, role: 'presentation', className: theme.image }),
    function () {
      if (caption) {
        return _react2.default.createElement(
          'figcaption',
          { className: theme.imageCaption },
          _react2.default.createElement(
            'p',
            null,
            caption
          )
        );
      }

      return null;
    }()
  );
};

UiImage.propTypes = {
  alt: _propTypes2.default.string.isRequired,
  caption: _propTypes2.default.string,
  src: _propTypes2.default.string.isRequired,
  theme: _propTypes2.default.shape({
    imageContainer: _propTypes2.default.string,
    image: _propTypes2.default.string,
    imageCaption: _propTypes2.default.string
  })
};

UiImage.defaultProps = {
  caption: '',
  theme: {}
};

exports.default = UiImage;
//# sourceMappingURL=UiImage.js.map