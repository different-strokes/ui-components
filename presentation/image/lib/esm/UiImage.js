import React from 'react';
import PropTypes from 'prop-types';

var UiImage = function UiImage(_ref) {
  var alt = _ref.alt,
      caption = _ref.caption,
      src = _ref.src,
      theme = _ref.theme;
  return React.createElement(
    'figure',
    { className: theme.imageContainer },
    React.createElement('img', { src: src, alt: alt, role: 'presentation', className: theme.image }),
    function () {
      if (caption) {
        return React.createElement(
          'figcaption',
          { className: theme.imageCaption },
          React.createElement(
            'p',
            null,
            caption
          )
        );
      }

      return null;
    }()
  );
};

UiImage.propTypes = {
  alt: PropTypes.string.isRequired,
  caption: PropTypes.string,
  src: PropTypes.string.isRequired,
  theme: PropTypes.shape({
    imageContainer: PropTypes.string,
    image: PropTypes.string,
    imageCaption: PropTypes.string
  })
};

UiImage.defaultProps = {
  caption: '',
  theme: {}
};

export default UiImage;
//# sourceMappingURL=UiImage.js.map