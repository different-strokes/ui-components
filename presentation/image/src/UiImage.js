import React from 'react';
import PropTypes from 'prop-types';

const UiImage = ({ alt, caption, src, theme }) => (
  <figure className={theme.imageContainer}>
    <img src={src} alt={alt} role="presentation" className={theme.image} />
    {(() => {
      if (caption) {
        return (
          <figcaption className={theme.imageCaption}>
            <p>{caption}</p>
          </figcaption>
        );
      }

      return null;
    })()}
  </figure>
);

UiImage.propTypes = {
  alt: PropTypes.string.isRequired,
  caption: PropTypes.string,
  src: PropTypes.string.isRequired,
  theme: PropTypes.shape({
    imageContainer: PropTypes.string,
    image: PropTypes.string,
    imageCaption: PropTypes.string,
  }),
};

UiImage.defaultProps = {
  caption: '',
  theme: {},
};

export default UiImage;
