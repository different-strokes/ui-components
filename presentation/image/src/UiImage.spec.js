import React from 'react';
import renderer from 'react-test-renderer';

import UiImage from './UiImage';

describe('UiImage', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <UiImage />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
