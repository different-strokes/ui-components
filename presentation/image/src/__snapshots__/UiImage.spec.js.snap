// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiImage renders correctly 1`] = `
<figure
  className={undefined}
>
  <img
    alt={undefined}
    className={undefined}
    role="presentation"
    src={undefined}
  />
</figure>
`;
